### PR TITLE
Added incident report for latest v0.29 out of ordering artefact pubishing

### PR DESCRIPTION
**Pull request by enekofb**
Part of https://github.com/weaveworks/weave-gitops/issues/3907

Documents the incident and raised to get feedback on the prevention actions 

**What changed?**

Added incident report for issue https://github.com/weaveworks/weave-gitops/issues/3907

### CODE REVIEW

File: doc/incidents/issues-3907/README.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+Protection against merging unapproved PRs is already in place.
+
+![weave-gitops-protection-approval.png](imgs/weave-gitops-protection-approval.png)
+
+Protection against merging not-released PRs is not in place:  
+
+![weave-gitops-protection-status-checks.png](imgs/weave-gitops-protection-status-checks.png)
+
+We could add that layer of protection by leveraging [github status checks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks)
+and requiring a `release` status check to merge PRs:
```
Comment (bigkevmcd) [inline doc/incidents/issues-3907/README.md:151]: Nice idea!

**Review by bigkevmcd — APPROVED** [review]
Let's give it a go.

File: doc/incidents/issues-3907/README.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+## Resolution 
+
+It was resolved after the release workflow ended and the container image was published.
+
+## Prevention
+
+Given that a Releaser just needs to verify the release looks fine before triggered. We should design it the process 
+with that into consideration. Two alternatives are identified: 
+
+1. no human intervention has effect or not possible after release PR is approved
```
Comment (hiddeco) [inline doc/incidents/issues-3907/README.md:113]:
```suggestion
1. no human intervention has effect, or is possible, after a release PR is approved
```

is what I think the idea is?

**Review by serboctor — APPROVED** [review]
LGTM

File: doc/incidents/issues-3907/README.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+2023-08-02T11:16:18.4302382Z   GCLOUD_PROJECT: docs-gitops-ww-prod
+2023-08-02T11:16:18.4302728Z   GCP_PROJECT: docs-gitops-ww-prod
+2023-08-02T11:16:18.4303069Z   GOOGLE_CLOUD_PROJECT: docs-gitops-ww-prod
+2023-08-02T11:16:18.4303393Z ##[endgroup]
+2023-08-02T11:16:19.8473627Z Pushed: ghcr.io/weaveworks/charts/weave-gitops:4.0.27
+2023-08-02T11:16:19.8474536Z Digest: sha256:014965fae24719c62faf864c6748f6c24437c6fc437d0920131765a78ca91650
+2023-08-02T11:16:19.8550603Z Post job cleanup.
+```
+
+First time we were notified of impact was at [11.26 UTC](https://weaveworks.slack.com/archives/CMDJFD3P0/p1690975563096289).
```
[the remark this answers is not in the document]
Comment (enekofb) [reply doc/incidents/issues-3907/README.md]: replaced with screenshot

File: doc/incidents/issues-3907/README.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    
+  Scenario: current flow  
+    Given a non release pr
+    When build
+    Then CI workflow  passes
+    And I could merge
+```
+
+
+```
```
[the remark this answers is not in the document]
Comment (enekofb) [reply doc/incidents/issues-3907/README.md:204]: yes, it tries to walk the solution in steps: first introduces the scenario, then walk the background.

**Pull request merged**